lang/cs/Org.Apache.REEF.Tang/Protobuf/ProtocolBufferClassHierarchy.cs

```diff
@@ -40,6 +40,8 @@ public class ProtocolBufferClassHierarchy : IClassHierarchy
         private readonly IDictionary<string, INode> lookupTable = new Dictionary<string, INode>();
         private readonly IDictionary<string, IDictionary<string, string>> _aliasLookupTable = new Dictionary<string, IDictionary<string, string>>();
 
+        private static object childIteratorLock = new object();
```

Comment:
Is it safe to use a static field as a lock? What if there are multiple instances of this same class at play, won't they suffer from (unneeded) contention on that same lock?